pkg/backingstore/reconciler.go

```
@@ -257,7 +257,7 @@ func (r *Reconciler) ReconcilePhases() error {
 
 // LoadBackingStoreSecret loads the secret to the reconciler struct
 func (r *Reconciler) LoadBackingStoreSecret() error {
-	if !util.IsSTSClusterBS(r.BackingStore) {
+	if !util.IsSTSClusterBS(r.BackingStore) {  //LMLM TODO: why is this negative condition, why not return nill if it is true?
```

If it is not an sts cluster and the secret is also not found it should return an error, And if it is an STS cluster it should not check for the secret.

---

The thing with negative conditions is that it is harder for humans to understand... 

The translation to a positive statement is in the second half of your sentence: 
> if it is an STS cluster it should not check for the secret.
Which can in turn be written like:
```
	if util.IsSTSClusterBS(r.BackingStore) {
		return nil
	}
```
@ezio-auditore  WDYT?

---

ok, makes sense.